shaders/CompoundCloudPlane.gdshader

```diff
@@ -5,58 +5,94 @@ render_mode unshaded;
 uniform sampler2D densities;
 uniform sampler2D noise;
 
-uniform vec4 colour1 : source_color = vec4(0, 0, 0, 0);
-uniform vec4 colour2 : source_color = vec4(0, 0, 0, 0);
-uniform vec4 colour3 : source_color = vec4(0, 0, 0, 0);
-uniform vec4 colour4 : source_color = vec4(0, 0, 0, 0);
+uniform vec4 colour1 : source_color = vec4(1.0, 1.0, 0.0, 1.0);
+uniform vec4 colour2 : source_color = vec4(0.0, 0.0, 1.0, 1.0);
+uniform vec4 colour3 : source_color = vec4(0.0, 1.0, 0.0, 1.0);
+uniform vec4 colour4 : source_color = vec4(1.0, 0.0, 0.0, 1.0);
 
 uniform vec2 UVOffset = vec2(0, 0);
 
-uniform float BrightnessMultiplier = 1.0f;
-uniform float CloudAlpha = 0.7f;
+uniform float BrightnessMultiplier = 0.10;
+uniform float NoiseScale = 0.1;
 
 // Setting this too low makes the clouds invisible
-const float CLOUD_DISSIPATION = 0.70f;
-
-const float DENSITY_MULTIPLIER = 0.95f;
+const float CLOUD_DISSIPATION = 0.70;
+const float DENSITY_MULTIPLIER = 0.95;
 
 // Should be the same as its counterpart in CompoundCloudPlane.cs
-const float CLOUD_MAX_INTENSITY_SHOWN = 1000.f;
+const float CLOUD_MAX_INTENSITY_SHOWN = 1000.0;
 
 // This needs to be less than 0.5 otherwise large cloud areas (due to the noise texture
 // not being very high resolution) are invisible
-const float NOISE_ZERO_OFFSET = 0.001f;
+const float NOISE_ZERO_OFFSET = 0.001;
+
+//You could be asking why there are two intensities values? One handles the cloud itself while the other has to take care of the noise itself
+//Probably could be further optimized
+
+const float NOISE_INTENSITY = 0.09;
+
+float getIntensity(float value) 
+{
+    return min(DENSITY_MULTIPLIER * atan(0.006 * CLOUD_MAX_INTENSITY_SHOWN * value), 1.0) * BrightnessMultiplier;
+}
 
-float getIntensity(float value){
-    return min(DENSITY_MULTIPLIER * atan(0.006f * CLOUD_MAX_INTENSITY_SHOWN * value), 1.0f) * BrightnessMultiplier;
+//Makes the noise sparse, i did my best to optimize the function
+
+float sparseNoise(vec2 uv, float scale, float contrast) 
+{
+    float noiseValue = texture(noise, uv * scale).r;
+    noiseValue = 0.5 + contrast * (noiseValue - 0.5);
+    
+    vec2 offsets[8] = vec2[]
+    (
+        vec2(1.0, 0.0),
+        vec2(-1.0, 0.0),
+        vec2(0.0, 1.0),
+        vec2(0.0, -1.0),
+        vec2(1.0, 1.0),
+        vec2(-1.0, -1.0),
+        vec2(1.0, -1.0),
+        vec2(-1.0, 1.0)
+    );
+    
+    float sum = noiseValue;
+    for (int i = 0; i < 8; i++) 
+    {
+        sum += texture(noise, uv + offsets[i] * scale).r;
```

Comment:
Is this where most of the performance impact comes from for these shader changes? I don't know that much about graphics programming but if this was possible to avoid by generating a noise texture ahead of time in some specific way, that would be much more performance friendly, right?